### PR TITLE
[INJIWEB-1756] Add URL wildcard check for datashare resource URL as per security team feedback

### PR DESCRIPTION
**Pull request by kongadurgesh**
Added validation for the datashare URL received as a input to download endpoint. The aim is to ensure that there is no script or illegal characters added to the URL before mimoto service tries fetch the resource from the datashare service.

## Summary by CodeRabbit

* **Bug Fixes**
  * Enhanced URL validation for data share credentials to enforce safety checks on resource paths and reject malformed URLs with proper error messaging.

* **Tests**
  * Added comprehensive test coverage for invalid credential resource URL scenarios including malformed paths and encoding edge cases.

### CODE REVIEW

**Review by jackjain — APPROVED** [review]

File: src/main/java/io/mosip/mimoto/service/impl/DataShareServiceImpl.java [hunk 1]
```diff
@@ -119,4 +129,49 @@ public  VCCredentialResponse downloadCredentialFromDataShare(PresentationRequest
...
+                    ErrorConstants.RESOURCE_INVALID.getErrorCode(),
+                    "Invalid path structure in resource URL");
+        }
+
+        String[] segments = decodedPath.split("/");
+        String wildcardPart;
+        if( segments.length == 0 ){
+            throw new InvalidCredentialResourceException(
+                    ErrorConstants.RESOURCE_INVALID.getErrorCode(),
+                    "Invalid resource identifier in URL");
```
[the remark this answers is not in the document]
Comment (jackjain) [reply src/main/java/io/mosip/mimoto/service/impl/DataShareServiceImpl.java:164]: @coderabbitai It doesn't look like "/".split("/") is evaluating into a non-empty array, can you recheck once with Java 21 ?
